pkg/volume/iscsi/iscsi_util.go

```diff
@@ -390,3 +407,31 @@ func extractPortalAndIqn(device string) (string, string, error) {
 	iqn := device[ind2:ind]
 	return portal, iqn, nil
 }
+
+func removeDuplicatePortal(bkpPortal []string) []string {
+	portals := make([]string, 0, len(bkpPortal))
+	found := map[string]bool{}
+	for _, tp := range bkpPortal {
+		if !found[tp] {
+			found[tp] = true
+			portals = append(portals, tp)
+		}
+	}
+	return portals
```

Comment:
My bad
I expected the len of the new array to be 3 and the last element empty in the case
of having a list of three portals with on set of duplicates

See
https://play.golang.org/p/hkm8_bhlOy